pkg/apis/operator/v1alpha1/tektontrigger_lifecycle.go

```diff
@@ -0,0 +1,141 @@
+/*
+Copyright 2020 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"knative.dev/pkg/apis"
+)
+
+var (
+	_ TektonComponentStatus = (*TektonTriggerStatus)(nil)
+
+	triggersCondSet = apis.NewLivingConditionSet(
+		DependenciesInstalled,
+		DeploymentsAvailable,
+		InstallSucceeded,
+		VersionMigrationEligible,
+	)
+)
+
+// GroupVersionKind returns SchemeGroupVersion of a TektonTrigger
+func (tp *TektonTrigger) GroupVersionKind() schema.GroupVersionKind {
+	return SchemeGroupVersion.WithKind(KindTektonTrigger)
+}
+
+// GetCondition returns the current condition of a given condition type
+func (tps *TektonTriggerStatus) GetCondition(t apis.ConditionType) *apis.Condition {
+	return triggersCondSet.Manage(tps).GetCondition(t)
+}
+
+// InitializeConditions initializes conditions of an TektonTriggerStatus
+func (tps *TektonTriggerStatus) InitializeConditions() {
+	triggersCondSet.Manage(tps).InitializeConditions()
+}
+
+// IsReady looks at the conditions returns true if they are all true.
+func (tps *TektonTriggerStatus) IsReady() bool {
+	return triggersCondSet.Manage(tps).IsHappy()
+}
+
+// MarkInstallSucceeded marks the InstallationSucceeded status as true.
+func (tps *TektonTriggerStatus) MarkInstallSucceeded() {
+	triggersCondSet.Manage(tps).MarkTrue(InstallSucceeded)
+	if tps.GetCondition(DependenciesInstalled).IsUnknown() {
+		// Assume deps are installed if we're not sure
+		tps.MarkDependenciesInstalled()
+	}
+}
+
+// MarkInstallFailed marks the InstallationSucceeded status as false with the given
+// message.
+func (tps *TektonTriggerStatus) MarkInstallFailed(msg string) {
+	triggersCondSet.Manage(tps).MarkFalse(
+		InstallSucceeded,
+		"Error",
+		"Install failed with message: %s", msg)
+}
+
+// MarkVersionMigrationEligible marks the VersionMigrationEligible status as false with given message.
+func (tps *TektonTriggerStatus) MarkVersionMigrationEligible() {
+	triggersCondSet.Manage(tps).MarkTrue(VersionMigrationEligible)
+}
+
+// MarkVersionMigrationNotEligible marks the DeploymentsAvailable status as true.
+func (tps *TektonTriggerStatus) MarkVersionMigrationNotEligible(msg string) {
+	triggersCondSet.Manage(tps).MarkFalse(
+		VersionMigrationEligible,
+		"Error",
+		"Version migration is not eligible with message: %s", msg)
+}
```

Comment:
@savitaashture these functions could cause some issues in the `status.conditions` update. I had to remove them to get the conditions to update properly. ref: https://github.com/tektoncd/operator/pull/157/files#diff-e979f9bfb2570f478176c1fde3598f51

shall we remove these from here as well. We put them back there when we add more update related features

Reply:
done